stellarphot/photometry/photometry.py

```diff
@@ -39,14 +42,72 @@
 EXPOSURE_KEYWORDS = ["EXPOSURE", "EXPTIME", "TELAPSE", "ELAPTIME", "ONTIME", "LIVETIME"]
 
 
+class AperturePhotometry(BaseModel):
+    """Class to perform aperture photometry on one or more images"""
+
+    settings: PhotometrySettings
+
+    @validate_call
+    def __call__(
+        self,
+        file_or_directory: str | Path,
+        **kwargs,
+    ) -> PhotometryData:
+        """
+        Perform aperture photometry on a single image or a directory of images.
+
+        Parameters
+        ----------
+        file_or_directory : str or Path
+            The file or directory on which to perform aperture photometry.
+
+        fname : str, optional (Default: None)
```

Comment:
Isn't `fname` redundant if the value of `file_or_directory` already automatically switches whether it is single image processing or multi-image processing based on it being a file or directory.  Do you want to support wildcard (e.g. glob) processing of filename?